jetstream.go

```diff
@@ -0,0 +1,334 @@
+package nats
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+)
+
+var (
+	AckAck      = []byte("+ACK")
+	AckNak      = []byte("-NAK")
+	AckProgress = []byte("+WPI")
+	AckNext     = []byte("+NXT")
+	AckTerm     = []byte("+TERM")
+)
+
+// JetStreamMsgMetaData is metadata related to a JetStream originated message
+type JetStreamMsgMetaData struct {
+	stream    string
```

Comment:
Why not just publicly accessible struct properties? I think that may be the preferred way here.

Reply:
Because idiomatic go is to return concrete types with accessors so people can easily test etc, maybe we make `Stream` and `GetStream()`, I'd like to do the same on Msg tbh

Reply:
I would just make the structs members public and not have any accessors etc.